syntax = "proto3";

package transaction_verification;

service TransactionVerification {
    rpc VerifyTransaction (TransactionVerificationRequest) returns (TransactionVerificationResponse);
}

message TransactionVerificationRequest {
    string transaction_id = 1;
    string user_id = 2;
    repeated string items = 3;
    string credit_card = 4;
}

message TransactionVerificationResponse {
    bool is_valid = 1;
    string message = 2;
}
